clients/src/main/java/org/apache/kafka/clients/admin/internals/ListConsumerGroupOffsetsHandler.java

```diff
@@ -0,0 +1,148 @@
+        return "offsetFetch";
+    }
+
+    @Override
+    public AdminApiLookupStrategy<CoordinatorKey> lookupStrategy() {
+        return lookupStrategy;
+    }
+
+    @Override
+    public OffsetFetchRequest.Builder buildRequest(int coordinatorId, Set<CoordinatorKey> keys) {
+        // Set the flag to false as for admin client request,
+        // we don't need to wait for any pending offset state to clear.
+        return new OffsetFetchRequest.Builder(groupId.idValue, false, partitions, false);
+    }
+
+    @Override
+    public ApiResult<CoordinatorKey, Map<TopicPartition, OffsetAndMetadata>> handleResponse(
+        Node coordinator,
+        Set<CoordinatorKey> groupIds,
+        AbstractResponse abstractResponse
+    ) {
+        final OffsetFetchResponse response = (OffsetFetchResponse) abstractResponse;
+        Map<CoordinatorKey, Map<TopicPartition, OffsetAndMetadata>> completed = new HashMap<>();
+        Map<CoordinatorKey, Throwable> failed = new HashMap<>();
+        List<CoordinatorKey> unmapped = new ArrayList<>();
+
+        if (response.error() != Errors.NONE) {
+            handleError(groupId, response.error(), failed, unmapped);
+        } else {
+            final Map<TopicPartition, OffsetAndMetadata> groupOffsetsListing = new HashMap<>();
+            for (Map.Entry<TopicPartition, OffsetFetchResponse.PartitionData> entry :
+                response.responseData().entrySet()) {
+                final TopicPartition topicPartition = entry.getKey();
+                OffsetFetchResponse.PartitionData partitionData = entry.getValue();
+                final Errors error = partitionData.error;
+
+                if (error == Errors.NONE) {
+                    final long offset = partitionData.offset;
+                    final String metadata = partitionData.metadata;
+                    final Optional<Integer> leaderEpoch = partitionData.leaderEpoch;
+                    // Negative offset indicates that the group has no committed offset for this partition
+                    if (offset < 0) {
+                        groupOffsetsListing.put(topicPartition, null);
+                    } else {
+                        groupOffsetsListing.put(topicPartition, new OffsetAndMetadata(offset, leaderEpoch, metadata));
+                    }
+                } else {
+                    log.warn("Skipping return offset for {} due to error {}.", topicPartition, error);
+                }
+            }
+            completed.put(groupId, groupOffsetsListing);
+        }
+        return new ApiResult<>(completed, failed, unmapped);
+    }
+
+    private void handleError(
+        CoordinatorKey groupId,
+        Errors error,
+        Map<CoordinatorKey,
+        Throwable> failed,
+        List<CoordinatorKey> unmapped
+    ) {
+        switch (error) {
+            case GROUP_AUTHORIZATION_FAILED:
+                log.error("Received authorization failure for group {} in `DeleteConsumerGroupOffsets` response", groupId,
```

Comment:
Seems like there's a copy/paste error here. I think we want `DeleteConsumerGroupOffsets` here to be `ListConsumerGroupOffsets`. Perhaps instead of writing the name for each one of these, we could just call `apiName()` instead?